src/corppa/utils/filter.py

```diff
@@ -97,13 +123,33 @@ def main():
     # progress bar is enabled by default; disable if requested
     disable_progress = not args.progress
 
-    output_filename = args.output
+    if not os.path.exists(args.idfile):
+        print(f"Error: idfile {args.idfile} does not exist")
+        exit(-1)
+    elif os.path.getsize(args.idfile) == 0:
+        print(f"Error: idfile {args.idfile} is empty")
+        exit(-1)
+
     # if requested output filename has no extension, add jsonl
+    output_filename = args.output
     if os.path.splitext(output_filename)[1] == "":
         output_filename = f"{output_filename}.jsonl"
-    save_filtered_corpus(
-        args.input, output_filename, args.idfile, disable_progress=disable_progress
-    )
+
+    if os.path.exists(output_filename):
+        print(
+            f"Error: requested output file {args.output} already exists; not overwriting"
+        )
+        exit(-1)
+
+    try:
+        save_filtered_corpus(
+            args.input, args.idfile, output_filename, disable_progress=disable_progress
+        )
+    except (FileNotFoundError, JSONDecodeError) as err:
+        # catch known possible errors and display briefly
+        # with the type of error and the brief message
+        print(f"{err.__class__.__name__}: {err}")
+        exit(-1)
```

Comment:
See comment for Line 128

```suggestion
        sys.exit(-1)
```